Error: `sudo: command not found` when deploying VMs

**Describe the bug**
I get the error (bash: line 1: sudo: command not found) when trying to create VMs.

**To Reproduce**
Steps to reproduce the behaviour:

1. Create a resource
2. Run 'terraform apply'
3. See error

Please also provide a minimal Terraform configuration that reproduces the issue.

```hcl
provider "proxmox" {
}
variable "proxmox_target_node" {
  type        = string
  description = "Proxmox target node"
  default     = "pve-1"
}
data "tls_public_key" "ssh_key" {
  private_key_openssh = file(pathexpand("~/.ssh/id_rsa"))
}
resource "proxmox_virtual_environment_file" "cloudinit_userdata" {
  content_type = "snippets"
  datastore_id = "local"
  node_name = var.proxmox_target_node
  source_file {
    path = "./files/user_data_ubuntu.yaml"
    checksum = sha256(file("./files/user_data_ubuntu.yaml"))
    file_name = "ubuntu.cloud-config.yaml"
  }
}
resource "proxmox_virtual_environment_file" "ubuntu_lunar_cloud_kvm_image" {
  content_type = "iso"
  datastore_id = "local"
  overwrite = false
  node_name    = var.proxmox_target_node

  source_file {
    path = "https://cloud-images.ubuntu.com/lunar/current/lunar-server-cloudimg-amd64-disk-kvm.img"
  }
}
resource "proxmox_virtual_environment_vm" "redis" {
  description = "Redis Server"
  node_name   = var.proxmox_target_node
  name        = "redis"
  tags        = ["redis"]
  agent {
    enabled = true
    trim    = true
  }
  disk {
    datastore_id = "local-lvm"
    file_id      = proxmox_virtual_environment_file.ubuntu_lunar_cloud_kvm_image.id
    interface    = "virtio0"
    size         = 15
    discard      = "on"
  }
  memory {
    dedicated = 512
  }
  operating_system {
    type = "l26"
  }
  initialization {
    ip_config {
      ipv4 {
        address = "dhcp"
      }
    }
    user_account {
      keys     = [trimspace(data.tls_public_key.ssh_key.public_key_openssh)]
      password = var.override_password
      username = "ubuntu"
    }
    user_data_file_id = proxmox_virtual_environment_file.cloudinit_userdata.id
  }
  lifecycle {
    ignore_changes = [description]
  }
  serial_device {}
  vga {
    enabled = true
    memory  = 16
    type    = "serial0"

  }
  network_device {
    bridge = "vmbr0"
  }
}

```

**Expected behavior**
Create the VMs with no errors

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Additional context**
Environment variables `PROXMOX_VE_USERNAME`, `PROXMOX_VE_PASS` and `PROXMOX_VE_ENDPOINT` are already set.

The workflow works fine on version 0.44.0 of this provider.

- Single or clustered Proxmox: Single Proxmox
- Provider version (ideally it should be the latest version): Latest 0.45.0
- Terraform version: Terraform v1.7.1
- OS (where you run Terraform from): MacOS
- Debug logs (`TF_LOG=DEBUG terraform apply`):
```
2024-01-26T11:43:42.373+0400 [DEBUG] provider.terraform-provider-proxmox_v0.45.0: SSH connection established: @module=proxmox user=root host=192.168.1.10:22 tf_mux_provider=tf5to6server.v5tov6Server tf_provider_addr=registry.terraform.io/bpg/proxmox tf_req_id=6ed73c1b-89d0-35fe-b46f-ce7d119819de @caller=github.com/bpg/terraform-provider-proxmox/proxmox/ssh/client.go:297 tf_resource_type=proxmox_virtual_environment_vm tf_rpc=ApplyResourceChange timestamp="2024-01-26T11:43:42.373+0400"
2024-01-26T11:43:42.646+0400 [ERROR] provider.terraform-provider-proxmox_v0.45.0: Failed to close: @caller=github.com/bpg/terraform-provider-proxmox/utils/io.go:20 error=EOF tf_provider_addr=registry.terraform.io/bpg/proxmox tf_req_id=6ed73c1b-89d0-35fe-b46f-ce7d119819de tf_resource_type=proxmox_virtual_environment_vm tf_rpc=ApplyResourceChange @module=proxmox tf_mux_provider=tf5to6server.v5tov6Server timestamp="2024-01-26T11:43:42.646+0400"
2024-01-26T11:43:42.646+0400 [ERROR] provider.terraform-provider-proxmox_v0.45.0: Response contains error diagnostic: @module=sdk.proto diagnostic_severity=ERROR tf_proto_version=6.4 tf_resource_type=proxmox_virtual_environment_vm tf_rpc=ApplyResourceChange @caller=github.com/hashicorp/terraform-plugin-go@v0.20.0/tfprotov6/internal/diag/diagnostics.go:62
  diagnostic_summary=
  | bash: line 1: sudo: command not found
   diagnostic_detail="" tf_provider_addr=registry.terraform.io/bpg/proxmox tf_req_id=6ed73c1b-89d0-35fe-b46f-ce7d119819de timestamp="2024-01-26T11:43:42.646+0400"
2024-01-26T11:43:42.647+0400 [ERROR] provider.terraform-provider-proxmox_v0.45.0: Failed to close: @caller=github.com/bpg/terraform-provider-proxmox/utils/io.go:20 error=EOF tf_mux_provider=tf5to6server.v5tov6Server tf_req_id=eb67bbc5-2f8e-d76d-8940-6c61ecd0b34b @module=proxmox tf_provider_addr=registry.terraform.io/bpg/proxmox tf_resource_type=proxmox_virtual_environment_vm tf_rpc=ApplyResourceChange timestamp="2024-01-26T11:43:42.647+0400"
2024-01-26T11:43:42.648+0400 [ERROR] provider.terraform-provider-proxmox_v0.45.0: Response contains error diagnostic: diagnostic_detail=""
  diagnostic_summary=
  | bash: line 1: sudo: command not found
   diagnostic_severity=ERROR tf_proto_version=6.4 tf_provider_addr=registry.terraform.io/bpg/proxmox @module=sdk.proto tf_resource_type=proxmox_virtual_environment_vm tf_rpc=ApplyResourceChange tf_req_id=eb67bbc5-2f8e-d76d-8940-6c61ecd0b34b @caller=github.com/hashicorp/terraform-plugin-go@v0.20.0/tfprotov6/internal/diag/diagnostics.go:62 timestamp="2024-01-26T11:43:42.648+0400"
2024-01-26T11:43:42.660+0400 [DEBUG] State storage *statemgr.Filesystem declined to persist a state snapshot
2024-01-26T11:43:42.660+0400 [ERROR] vertex "proxmox_virtual_environment_vm.redis" error: bash: line 1: sudo: command not found
2024-01-26T11:43:42.669+0400 [DEBUG] State storage *statemgr.Filesystem declined to persist a state snapshot
2024-01-26T11:43:42.669+0400 [ERROR] vertex "proxmox_virtual_environment_vm.gitlab" error: bash: line 1: sudo: command not found
╷
│ Error: bash: line 1: sudo: command not found
│ 
│ 
│   with proxmox_virtual_environment_vm.redis,
│   on databases.tf line 104, in resource "proxmox_virtual_environment_vm" "redis":
│  104: resource "proxmox_virtual_environment_vm" "redis" {
│ 
╵
╷
│ Error: bash: line 1: sudo: command not found
│ 
│ 
│   with proxmox_virtual_environment_vm.gitlab,
│   on gitlab.tf line 1, in resource "proxmox_virtual_environment_vm" "gitlab":
│    1: resource "proxmox_virtual_environment_vm" "gitlab" {
│ 
╵
2024-01-26T11:43:42.685+0400 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2024-01-26T11:43:42.687+0400 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/bpg/proxmox/0.45.0/darwin_arm64/terraform-provider-proxmox_v0.45.0 pid=21259
2024-01-26T11:43:42.687+0400 [DEBUG] provider: plugin exited
```

--- a/docs/index.md
+++ b/docs/index.md
@@ -136,6 +136,10 @@ This is because the provider needs to know which PAM user to use for the SSH con
 
 When using a non-root user for the SSH connection, the user **must** have the `sudo` privilege on the target node without requiring a password.
 
+-> If you run clustered Proxmox VE, you will need to configure the `sudo` privilege for the user on all nodes in the cluster.
+
+-> `sudo` is not installed by default on Proxmox VE nodes. You can install it via the command line on the Proxmox host: `apt install sudo`
+
 You can configure the `sudo` privilege for the user via the command line on the Proxmox host. In the example below, we create a user `terraform` and assign the `sudo` privilege to it:
 
 - Create a new system user:

--- a/proxmox/ssh/client.go
+++ b/proxmox/ssh/client.go
@@ -29,6 +29,9 @@ import (
 
 // Client is an interface for performing SSH requests against the Proxmox Nodes.
 type Client interface {
+	// Username returns the SSH username.
+	Username() string
+
 	// ExecuteNodeCommands executes a command on a node.
 	ExecuteNodeCommands(ctx context.Context, nodeName string, commands []string) ([]byte, error)
 
@@ -69,6 +72,10 @@ func NewClient(
 		agentSocket:  agentSocket,
 		nodeResolver: nodeResolver,
 	}, nil
+}
+
+func (c *client) Username() string {
+	return c.username
 }
 
 // ExecuteNodeCommands executes commands on a given node.

--- a/proxmoxtf/resource/vm.go
+++ b/proxmoxtf/resource/vm.go
@@ -2974,6 +2974,7 @@ func vmCreateCustomDisks(ctx context.Context, d *schema.ResourceData, m interfac
 		commands = append(
 			commands,
 			`set -e`,
+			`try_sudo(){ if [ $(sudo -n echo tfpve 2>&1 | grep "tfpve" | wc -l) -gt 0 ]; then sudo $1; else $1; fi }`,
 			fmt.Sprintf(`file_id="%s"`, fileID),
 			fmt.Sprintf(`file_format="%s"`, fileFormat),
 			fmt.Sprintf(`datastore_id_target="%s"`, datastoreID),
@@ -2982,11 +2983,11 @@ func vmCreateCustomDisks(ctx context.Context, d *schema.ResourceData, m interfac
 			fmt.Sprintf(`disk_interface="%s"`, diskInterface),
 			fmt.Sprintf(`file_path_tmp="%s"`, filePathTmp),
 			fmt.Sprintf(`vm_id="%d"`, vmID),
-			`source_image=$(sudo pvesm path "$file_id")`,
-			`imported_disk="$(sudo qm importdisk "$vm_id" "$source_image" "$datastore_id_target" -format $file_format | grep "unused0" | cut -d ":" -f 3 | cut -d "'" -f 1)"`,
+			`source_image=$(try_sudo "pvesm path $file_id")`,
+			`imported_disk="$(try_sudo "qm importdisk $vm_id $source_image $datastore_id_target -format $file_format" | grep "unused0" | cut -d ":" -f 3 | cut -d "'" -f 1)"`,
 			`disk_id="${datastore_id_target}:$imported_disk${disk_options}"`,
-			`sudo qm set "$vm_id" "-${disk_interface}" "$disk_id"`,
-			`sudo qm resize "$vm_id" "${disk_interface}" "${disk_size}G"`,
+			`try_sudo "qm set $vm_id -${disk_interface} $disk_id"`,
+			`try_sudo "qm resize $vm_id ${disk_interface} ${disk_size}G"`,
 		)
 
 		importedDiskCount++
@@ -3006,6 +3007,11 @@ func vmCreateCustomDisks(ctx context.Context, d *schema.ResourceData, m interfac
 
 		out, err := api.SSH().ExecuteNodeCommands(ctx, nodeName, commands)
 		if err != nil {
+			if strings.Contains(err.Error(), "pvesm: not found") {
+				return diag.Errorf("The configured SSH user '%s' does not have the required permissions to import disks. "+
+					"Make sure `sudo` is installated and the user is a member of sudoers.", api.SSH().Username())
+			}
+
 			return diag.FromErr(err)
 		}
 

